architecture/2025-04-aci-confidential-containers.md

```diff
@@ -0,0 +1,388 @@
+# Radius Integration with Azure Container Instances (ACI) - Confidential Containers
+
+* **Author**: Shruthi Kumar (@sk593)
+
+## Overview
+
+The integration of Azure Container Instances (ACI) with Radius aims to enable deployments of applications and containers using the Radius control plane in Kubernetes to target ACI environments. One of the key features we want to implement is the use of confidential containers to enhance data security and privacy. Confidential containers on Azure provide a set of features and capabilities to further secure your standard container workloads. They run in a hardware-backed Trusted Execution Environment (TEE) that provides intrinsic capabilities like data integrity, data confidentiality, and code integrity.
+
+Confidential containers require a CCE policy to be set on the container group profile associated with ACI instances. CCE policies can only be generated using an extension on the Azure CLI. As a result, Radius will need access to the Azure CLI and the `confcom` extension to support this feature. More information on confidential containers with ACI can be found [here](https://learn.microsoft.com/en-us/azure/container-instances/container-instances-tutorial-deploy-confidential-containers-cce-arm)
```

Comment:
Could this be a deal breaker for some potential users because they may think that it is a vendor lock-in? Can users opt-out of this option or is it going to come out of the box when you run `rad init` for example?

Reply:
I don't think this would be a vendor lock-in. Users would opt-in to use ACI by specifying a different compute than Kubernetes.